extensions/ql-vscode/src/view/data-extensions-editor/DataExtensionsEditor.tsx

```diff
@@ -304,23 +296,21 @@ export function DataExtensionsEditor({
 
           <EditorContainer>
             <ButtonsContainer>
-              <VSCodeButton onClick={onSaveAllClick}>Apply</VSCodeButton>
+              <VSCodeButton
+                onClick={onSaveAllClick}
+                disabled={modifiedSignatures.size === 0}
+              >
+                Save all
+              </VSCodeButton>
               {viewState.enableFrameworkMode && (
                 <VSCodeButton appearance="secondary" onClick={onRefreshClick}>
                   Refresh
                 </VSCodeButton>
               )}
-              <VSCodeButton onClick={onGenerateFromSourceClick}>
-                {viewState.mode === Mode.Framework
-                  ? "Generate"
-                  : "Download and generate"}
-              </VSCodeButton>
-              {viewState.showLlmButton && (
-                <>
-                  <VSCodeButton onClick={onGenerateAllFromLlmClick}>
-                    Generate using LLM
-                  </VSCodeButton>
-                </>
+              {viewState.mode === Mode.Framework && (
+                <VSCodeButton onClick={onGenerateFromSourceClick}>
```

Comment:
I've just removed Application mode from the conditional. Should we also remove the references to Application mode from the underlying function [here](https://github.com/github/vscode-codeql/blob/c7b556e7487d222740cf6d666624709e9203e41a/extensions/ql-vscode/src/data-extensions-editor/data-extensions-editor-view.ts#L291), or are we expecting to use that somewhere else? 🤔

Reply:
Actually, ignore me! It looks like this function is also called from the "Model from source" button in application mode, so will keep it as-is: 
![image](https://github.com/github/vscode-codeql/assets/42641846/fcae4352-dbb0-41e6-87e9-4e570cc5318e)